File: GoaUni.py
import sys

class Student:
  def __init__(self, student_id, name):
    self.student_id = student_id
    self.name = name
    self.enrolled_courses = []  # List of course names

  def get_student_id(self):
    return self.student_id

  def get_name(self):
    return self.name

  def get_enrolled_courses(self):
    return self.enrolled_courses  # Direct access (not recommended)

  def enroll_in_course(self, course_name):
    self.enrolled_courses.append(course_name)

class GoaUniversity:
  def __init__(self, university_name):
    self.university_name = university_name
    self.enrolled_students = []
    self.offered_courses = []  # List of course names (optional)

  def get_university_name(self):
    return self.university_name

  def get_enrolled_students(self):
    return self.enrolled_students  # Direct access (not recommended)

  def add_student(self, student_id, name):
    student = Student(student_id, name)
    self.enrolled_students.append(student)

  def add_course(self, course_name):
    self.offered_courses.append(course_name)  # Optional

  def enroll_student_in_course(self, student_id, course_name):
    for student in self.enrolled_students:
      if student.student_id == student_id:
        student.enroll_in_course(course_name)
        return
    print(f"Student with ID {student_id} not found.")

  def display_all_information(self):
    print(f"Goa University")
    print("----------------")
    for student in self.enrolled_students:
      print(f"Student ID: {student.student_id}")
      print(f"Name: {student.name}")
      print(f"Enrolled Courses:")
      if not student.enrolled_courses:
        print("  - None")
      else:
        for course in student.enrolled_courses:
          print(f"  - {course}")
      print()

def main():
  goa_university = GoaUniversity("Goa University")

  while True:
    print("1. Add Student")
    print("2. Enroll Student in Course")
    print("3. Display All Information")
    print("4. Exit")
    print("Choose an option: ", end="")

    try:
      choice = int(input())
    except ValueError:
      print("Invalid choice. Please enter a number.")
      continue

    if choice not in range(1, 5):
      print("Invalid choice.")
      continue

    if choice == 1:
      student_id = input("Enter student ID: ")
      name = input("Enter student name: ")
      goa_university.add_student(student_id, name)
    elif choice == 2:
      student_id = input("Enter student ID: ")
      course_name = input("Enter course name: ")
      goa_university.enroll_student_in_course(student_id, course_name)
    elif choice == 3:
      goa_university.display_all_information()
    else:
      print("Exiting...")
      sys.exit(0)

if __name__ == "__main__":
  main()

File: GoaUni.cs
using System;
using System.Collections.Generic;
using System.Scanner; // Replace with appropriate input library (e.g., Console)

class Student
{
    public string StudentId { get; set; }
    public string Name { get; set; }
    public List<string> EnrolledCourses { get; private set; } // Consider using private setter for better encapsulation

    public Student(string studentId, string name)
    {
        this.StudentId = studentId;
        this.Name = name;
        this.EnrolledCourses = new List<string>();
    }

    public void EnrollInCourse(string courseName)
    {
        EnrolledCourses.Add(courseName);
    }
}

class GoaUniversity
{
    public string UniversityName { get; set; }
    public List<Student> EnrolledStudents { get; private set; } // Consider using private setter for better encapsulation
    public List<string> OfferedCourses { get; set; } // Optional

    public GoaUniversity(string universityName)
    {
        this.UniversityName = universityName;
        this.EnrolledStudents = new List<Student>();
        this.OfferedCourses = new List<string>();
    }

    public void AddStudent(string studentId, string name)
    {
        Student student = new Student(studentId, name);
        EnrolledStudents.Add(student);
    }

    public void AddCourse(string courseName)
    {
        OfferedCourses.Add(courseName);
    }

    public void EnrollStudentInCourse(string studentId, string courseName)
    {
        foreach (Student student in EnrolledStudents)
        {
            if (student.StudentId == studentId)
            {
                student.EnrollInCourse(courseName);
                return;
            }
        }
        Console.WriteLine($"Student with ID {studentId} not found.");
    }

    public void DisplayAllInformation()
    {
        Console.WriteLine($"Goa University");
        Console.WriteLine("----------------");
        foreach (Student student in EnrolledStudents)
        {
            Console.WriteLine($"Student ID: {student.StudentId}");
            Console.WriteLine($"Name: {student.Name}");
            Console.WriteLine("Enrolled Courses:");
            if (!student.EnrolledCourses.Any())
            {
                Console.WriteLine("  - None");
            }
            else
            {
                foreach (string course in student.EnrolledCourses)
                {
                    Console.WriteLine($"  - {course}");
                }
            }
            Console.WriteLine();
        }
    }
}

class GoaUniversityMain
{
    static void Main(string[] args)
    {
        GoaUniversity goaUniversity = new GoaUniversity("Goa University");

        while (true)
        {
            Console.WriteLine("1. Add Student");
            Console.WriteLine("2. Enroll Student in Course");
            Console.WriteLine("3. Display All Information");
            Console.WriteLine("4. Exit");
            Console.Write("Choose an option: ");

            int choice;
            try
            {
                choice = int.Parse(Console.ReadLine());
            }
            catch (FormatException)
            {
                Console.WriteLine("Invalid choice. Please enter a number.");
                continue;
            }

            if (choice < 1 || choice > 4)
            {
                Console.WriteLine("Invalid choice.");
                continue;
            }

            switch (choice)
            {
                case 1:
                    Console.Write("Enter student ID: ");
                    string studentId = Console.ReadLine();
                    Console.Write("Enter student name: ");
                    string name = Console.ReadLine();
                    goaUniversity.AddStudent(studentId, name);
                    break;
                case 2:
                    Console.Write("Enter student ID: ");
                    studentId = Console.ReadLine();
                    Console.Write("Enter course name: ");
                    string courseName = Console.ReadLine();
                    goaUniversity.EnrollStudentInCourse(studentId, courseName);
                    break;
                case 3:
                    goaUniversity.DisplayAllInformation();
                    break;
                case 4:
                    Console.WriteLine("Exiting...");
                    Environment.Exit(0);
                default:
                    break;
            }
        }
    }
}